core/coreapi/interface/util.go

```diff
@@ -1,10 +1,20 @@
 package iface
 
 import (
+	"context"
 	"io"
 )
 
 type Reader interface {
-	io.ReadSeeker
+	ReadSeekCloser
+	Size() uint64
+	CtxReadFull(context.Context, []byte) (int, error)
+}
+
+// A ReadSeekCloser implements interfaces to read, copy, seek and close.
+type ReadSeekCloser interface {
```

Comment:
This code seems to be repeating the interfaces already defined in `go-unixfs`:

https://github.com/ipfs/go-unixfs/blob/master/io/dagreader.go#L22-L37

(Or maybe it's the other way around and UnixFS is duplicating these ones, either way, is it possible for one to reference the other?)

Reply:
That will be doable once the coreapi interface is extracted, which I hope will happen fairly soon - for progress see https://github.com/ipfs/go-ipfs/issues/4498 (I want to have pubsub/swarm/unixfs done before extracting the interface)